src/libratbag-hidraw.h

```diff
@@ -183,6 +183,33 @@ int ratbag_hidraw_read_input_report(struct ratbag_device *device, uint8_t *buf,
  */
 int ratbag_hidraw_read_input_report_index(struct ratbag_device *device, uint8_t *buf, size_t len, int hidrawno);
 
+/**
+ * Read an input report from the device util a specified filter function is successfull
+ *
+ * @param device the ratbag device
+ * @param[out] buf resulting raw data
+ * @param len length of buf
+ * @param filter filter function
```

Comment:
that one needs documentation because just looking at this I wouldn't know what to return for actual filtered data.